src/commands/rps.ts

```diff
@@ -0,0 +1,193 @@
+    }
+
+    if (text === 'end') {
+      if (!this.challenger || this.challenger.id !== user.id) {
+        return 'You aren\'t challenging anyone!';
+      }
+      this.clear_game();
+      return user.username + ' has ended their challenge because they are a big, fat :chicken:';
+    }
+
+    return 'Usage challenge/accept/end';
+  }
+
+  @On('messageCreate')
+  rps_dm([message]: ArgsOf<'messageCreate'>) {
+    // Only accept messages that are DMs if a game of rps is ongoing
+    if (message.channel.type !== 'DM') {
+      return;
+    }
+    if (!this.challenger || !this.acceptor) {
+      return;
+    }
+
+    let player = null;
+    if (this.challenger.id === message.author.id) {
+      player = this.challenger;
+    }
+    if (this.acceptor.id === message.author.id) {
+      player = this.acceptor;
+    }
+    if (!player) {
+      return;
+    }
+
+    // Cannot change play
+    if (this.plays[player.id]) {
+      return;
+    }
+    if (message.content !== 'rock' && message.content !== 'paper' && message.content !== 'scissors') {
+      this.expect_play(player);
+    }
+
+    this.plays[player.id] = message.content;
+
+    if (!this.plays[this.challenger.id] || !this.plays[this.acceptor.id]) {
+      return;
+    }
+
+    // Both players have played - decide winner
+    let winner = null;
+    if (this.wins_table[this.plays[this.challenger.id]] === this.plays[this.acceptor.id]) {
+      winner = this.challenger;
+    }
+    if (this.wins_table[this.plays[this.acceptor.id]] === this.plays[this.challenger.id]) {
+      winner = this.acceptor;
+    }
+
+    if (!this.channel) {
+      console.log('Impossible: rps: this.channel not set');
+      return;
+    }
+    if (!winner) {
+      this.channel.send(this.challenger.username + ' and ' + this.acceptor.username + ' both show ' + this.plays[this.challenger.id] + '. It\'s a tie!');
+    }
+    else {
+      this.channel.send(this.challenger.username + ' shows ' + this.plays[this.challenger.id] + '. ' + this.acceptor.username + ' shows ' + this.plays[this.acceptor.id] + '.\n' + winner.username + ' wins!');
+    }
+
+    this.clear_game();
+  }
+  
+  @SimpleCommand('rps')
+  rps(
+    @SimpleCommandOption('text', { type: 'STRING' })
+    text: string,
+    command : SimpleCommandMessage
+  ) {
+    if (command.message.channel.id !== this.general_channel) {
+      return;
+    }
+    this.channel = command.message.channel;
+    this.channel.send(this.do_rps(text, command.message.author, command.message.channel));
+  }
+
+  @Slash('rps', { description: 'challenge, accept, or end a rock paper scissors game' })
+  async slash(
+    @SlashOption('text', { type: 'STRING' })
+    text: string,
+    interaction : CommandInteraction
+  ) {
+    if (!interaction.channel) {
+      return;
+    }
+    if (interaction.channel.id !== this.general_channel) {
+      return;
+    }
+    this.channel = interaction.channel;
+    interaction.reply(this.do_rps(text, interaction.user, interaction.channel));
```

Comment:
Should make sure to do `toLowerCase()` or `toUpperCase()` so the user can input CHalLenGE or whatever and it will still work.
```suggestion
    interaction.reply(this.do_rps(text.toLowerCase(), interaction.user, interaction.channel));
```